### ISSUE TITLE
False error on placeholder function

### ISSUE THREAD

**Issue by shellscape** [tonyganch/gonzales-pe#98]
This one is a _major blocker_ for us: gozales-pe throws an error when it encounters LESS placeholder functions. These are perfectly valid and pass linting with `lessc --lint`.

Test Script:

```
var gonzales = require('gonzales-pe');
var less = '.transition(...){}';
var result = gonzales.parse(less, {
    syntax: 'less'
});
console.log(result);
```

Output:

```
→ node lesstest.js                  

/.../node_modules/gonzales-pe/lib/parse.js:5
= parse(tokens,rule,needInfo);}catch(e) {if(!e.syntax)throw e;throw new Parsin
                                                                    ^
Parsing error: Please check validity of the block starting from line #1
```

When `var less = '.transition(...){}';` is changed to `var less = '.transition(){}';` gonzales-pe allows it to pass. That should not be required, as the former value is completely valid.

**Comment by tonyganch** [comment]
@shellscape, thank you for report!

Link to LESS docs: http://lesscss.org/features/#mixins-parametric-feature-advanced-arguments-and-the-rest-variable

**Comment by shellscape** [comment]
@tonyganch indeed, that's what I had referenced before hand. are you verifying that this is a bug?

**Comment by tonyganch** [comment]
@shellscape, that's a language feature I've had no idea about, so yes, you can call it a bug :)
I've left the link to LESS docs for future me, so I don't need to search it again once I'm ready to fix the issue.

**Comment by shellscape** [comment]
Great, thanks. I took a look at the source, but I'm a little lost at where to begin looking for how to fix this. Totally willing to put together a PR, but I need some direction I'm afraid.

**Comment by tonyganch** [comment]
You may find this pr useful: https://github.com/tonyganch/gonzales-pe/pull/78
There are some comments on how the parser works.

`checkArgument` function checks for all possible types of arguments: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L287
You can see that it also checks for a `variablesList` node which is `@rest...` in LESS: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L292

Less calls this `...` thing a variable dictionary, I call it a variables list: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L3170
Right now I assume that there is always a variable before `...`: it can be `@rest...` or `@panda...`.
The node for such a list looks like this:

```
{
  type: 'variablesList', // <= `@panda...`
  content: [{
    type: 'variable',    // <= `@panda`
    content: [{
      type: 'indent',    // <= `panda`
      content: 'panda'
    }]
  }]
}
```

What should be done is we should make a variable in variables list optional.
So `...` gets parsed as:

```
{
  type: 'variablesList', // <= `...`
  content: []
}
```

TL;DR:
(1) This line should be removed: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L3177
(2) And here you should check for a variable first and add it to the content only if it exists: https://github.com/tonyganch/gonzales-pe/blob/dev/src/less/parse.js#L3194

**Comment by shellscape** [comment]
@tonyganch I'm still seeing this error with 3.2.1

**Comment by tonyganch** [comment]
Well, the issue is still open, yes.
I have not touched it because of your comment: «Totally willing to put together a PR, but I need some direction I'm afraid».

**Comment by shellscape** [comment]
@tonyganch I've not been able to get anything working, hence no PR has been filed. The structure of the project has made it really difficult to work with, so I'm afraid I haven't made any progress. Please do proceed with a fix.

**Comment by shellscape** [comment]
Figured out a decent way to work with the lib. Still not ideal, but I managed to put together a few pull requests for fixes.

**Comment by simoami** [comment]
any updates on this? Even with external help this is not moving much.

**Comment by shellscape** [comment]
@simoami we've opted to switch from gonzales-pe to postcss + postcss-less
